src/test/integration/tests/Deposit_Delegate_Allocate.t.sol

```diff
@@ -27,21 +30,161 @@ contract Integration_Deposit_Delegate_Allocate is IntegrationCheckUtils {
 
         // 2. Deposit into strategies
         staker.depositIntoEigenlayer(strategies, tokenBalances);
-        uint[] memory shares = _calculateExpectedShares(strategies, tokenBalances);
+        uint256[] memory shares = _calculateExpectedShares(strategies, tokenBalances);
 
         // Check that the deposit increased operator shares the staker is delegated to
         check_Deposit_State(staker, strategies, shares);
         assert_Snap_Added_OperatorShares(operator, strategies, shares, "operator should have received shares");
 
         operator.registerForOperatorSets(operatorSets);
-        
-        for (uint i; i < operatorSets.length; ++i) {
+
+        AllocateParams[] memory allocateParams = new AllocateParams[](operatorSets.length);
+
+        for (uint256 i; i < operatorSets.length; ++i) {
             uint256 len = allocationManager.getStrategiesInOperatorSet(operatorSets[i]).length;
-            operator.modifyAllocations(operatorSets[i], _randMagnitudes({ sum: 1 ether / uint64(operatorSets.length), len: len }));
-            (IStrategy[] memory strategiesToSlash, uint[] memory wadsToSlash) = _randStrategiesAndWadsToSlash(operatorSets[i]);
-            avs.slashOperator(operator, operatorSets[i].id, strategiesToSlash, wadsToSlash);
+            allocateParams[i] = operator.modifyAllocations(
+                operatorSets[i], _randMagnitudes({sum: 1 ether / uint64(operatorSets.length), len: len})
+            );
+            assert_Snap_Allocations_Modified(
+                operator, allocateParams[i], false, "operator allocations should be updated before delay"
+            );
         }
 
-        // TODO: write checks for slashing...
+        _rollBlocksForCompleteAllocation();
+
+        for (uint256 i; i < operatorSets.length; ++i) {
+            assert_Snap_Allocations_Modified(
+                operator, allocateParams[i], true, "operator allocations should be updated after delay"
+            );
+        }
     }
-}
\ No newline at end of file
+
+    function testFuzz_deposit_delegate_allocate_slash_undelegate_completeAsTokens(
```

Comment:
To clarify, this is and the other slash + completeAsTokens integration tests failed due to the EigenPod gwei bug correct?

Reply:
yes